roles/repositories_sync/tasks/main.yml

```diff
@@ -0,0 +1,29 @@
+---
+- name: Sync repository
+  theforeman.foreman.repository_sync:
+    username: "{{ username | default(omit) }}"
+    password: "{{ password | default(omit) }}"
+    server_url: "{{ server_url | default(omit) }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    product: "{{ item.0.name }}"
+    repository: "{{ item.1.name }}"
+  with_subelements:
+    - "{{ products | selectattr('repository_sets', 'defined') | list }}"
```

Comment:
this means it won't sync any custom repositories that the `repositories` role can create.